src/client/js/Panels/MetaEditor/MetaEditorControl.js

```diff
@@ -1174,46 +1174,71 @@ define(['js/logger',
     };
 
     MetaEditorControl.prototype._createPointerRelationship = function (sourceID, targetID, isSet) {
-        var sourceNode = this._client.getNode(sourceID),
-            targetNode = this._client.getNode(targetID),
+        var client = this._client,
+            sourceNode = client.getNode(sourceID),
+            targetNode = client.getNode(targetID),
+            metaNodes = client.getAllMetaNodes(true),
             pointerMetaDescriptor,
+            existingNames,
             existingPointerNames,
-            notAllowedPointerNames,
-            self = this;
+            notAllowedPointerNames;
+
+        function getExistingNames() {
+            var nodeObj = sourceNode,
+                lastMetaNode,
+                i,
+                instancePaths;
+
+            // Get the meta-node "last" in the inheritance chain
+            do {
+                instancePaths = nodeObj.getInstancePaths();
+
+                lastMetaNode = null;
+                for (i = 0; i < instancePaths.length; i += 1) {
+                    if (metaNodes[instancePaths[i]]) {
+                        lastMetaNode = metaNodes[instancePaths[i]];
+                        break;
+                    }
+                }
+
+                if (lastMetaNode) {
+                    nodeObj = lastMetaNode;
+                }
+            } while (lastMetaNode);
+
+            // TODO: Currently these contain mixins too
+            return {
+                pointers: nodeObj.getValidPointerNames() || [],
+                sets: nodeObj.getValidSetNames() || [],
+                aspects: nodeObj.getValidAspectNames() || []
+            };
+        }
 
         if (sourceNode && targetNode) {
-            notAllowedPointerNames = _.union(sourceNode.getSetNames(), sourceNode.getPointerNames());
+            existingNames = getExistingNames();
+
             if (isSet === true) {
-                //this is a pointer list
-                existingPointerNames = _.difference(sourceNode.getSetNames() || [],
-                    sourceNode.getValidAspectNames());
+                //this is a set
+                existingPointerNames = existingNames.sets;
+                notAllowedPointerNames = _.union(existingNames.pointers, existingNames.aspects);
```

Comment:
yeah,  I think in theory aspects could use the same names, but it might be that on some level they share the same ids inside the node...